### PR TITLE
SVA-90 - Amended script.js to pull back data from two projects

### PR DESCRIPTION
**Pull request by CatMac228**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Checklist](#checklist)

# Description

- Amended Script.js to pull back data from the two Jira projects using Promise.all(). 
- Code is not fully working - it attempts to pull back the same data from both projects but unsure how to fix this - any feedback welcome!

https://sta2020.atlassian.net/browse/SVA-90 (originally SVA-64)

## Type of change

Please delete options that are not relevant
- [] Bug fix (non-breaking change which fixes an isssue)
- [x] New Feature (non-breaking change which adds functionlity)
- [] Breaking change(fix or feature that would cause existing functionlity to not work as expected)
- [] This change requires a documentation update

# Checklist

- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, pariculary in hard-to-understand areas
- [] I have made corresponding changes to the documentation (if required)
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by gunnx** [comment]
Try like this style: 

```
const fetchAll = async (urls) => {
  const responses = await Promise.all(urls.map(u => fetch(u)))
  const jsonPayloads = await Promise.all(responses.map(r => r.json()))
  console.log(jsonPayloads)
}

const urls = [
  'https://url1/api',
  'https://url2/api',
  'https://url3/api'
]
fetchAll(urls)
```

**Comment by CatMac228** [comment]
> const fetchAll = async (urls) => {
>   const responses = await Promise.all(urls.map(u => fetch(u)))
>   const jsonPayloads = await Promise.all(responses.map(r => r.json()))
>   console.log(jsonPayloads)
> }
> 
> const urls = [
>   'https://url1/api',
>   'https://url2/api',
>   'https://url3/api'
> ]
> fetchAll(urls)

Hi Keith, thanks for looking at this. I have tried your changes although I'm unsure where to pass in the headers for the requests without it looping 4 times & attempting to return the same data from both projects...any ideas how to resolve this? Headers are the same for both URL requests - cheers!

**Comment by gunnx** [comment]
you can pass the headers the same way you did in yours as a second parameter to fetch

```
const fetchOptions = { 
  headers: {
          'Authorization': `Basic ${Buffer.from(
                `tech_admin@scottishtecharmy.org:qSV1Uf9CJBM4NM3nl4ZX50F8`
              ).toString('base64')}`,
              'Accept': 'application/json'
            }
}
...
const responses = await Promise.all(urls.map(u => fetch(u, fetchOptions)))
```

**Comment by gunnx** [comment]
Also suggest you do not commit the authorisation value to github, and instead use Environment variables

**Comment by keithdhd** [comment]
The code looks good but I don't have the API key and email address to test it. Are these available somewhere or can I go ahead and approve this PR?

**Review by keithdhd — COMMENTED** [review]
The code looks good but I don't have the API key and email address to test it. Are these available somewhere or can I go ahead and approve this PR?

**Comment by CatMac228** [comment]
> The code looks good but I don't have the API key and email address to test it. Are these available somewhere or can I go ahead and approve this PR?

Will send them over to you 👍

**Comment by joanna-allen** [comment]
@ckalika could we enlist your help with this?

File: api/package.json [hunk 1]
```diff
@@ -11,10 +11,11 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "axios": "^0.23.0",
+    "axios": "^0.25.0",
     "dotenv": "^10.0.0",
     "express": "^4.17.1",
     "i": "^0.3.7",
+    "n": "^8.0.2",
```
Comment (gunnx) [inline api/package.json:18]: should this be here?

File: api/routes/projects.js [hunk 2]
```diff
@@ -1,9 +1,103 @@
...
+          Authorization: `Basic ${Buffer.from(
+            // below use email address you used for jira and generate token from jira
+            `${email}:${api_key}`,
+          ).toString('base64')}`,
+          Accept: 'application/json',
+        },
+      },
+    );
+
+    let ResTotalData = parseInt(jiraRes.data.total);
```
Comment (joefhall) [inline api/routes/projects.js:38]: I think this can be `const` rather than `let` as it isn't reassigned

File: api/routes/projects.js [hunk 3]
```diff
@@ -1,9 +1,103 @@
...
+          Authorization: `Basic ${Buffer.from(
+            // below use email address you used for jira and generate token from jira
+            `${email}:${api_key}`,
+          ).toString('base64')}`,
+          Accept: 'application/json',
+        },
+      },
+    );
+
+    let ItTotalData = parseInt(jiraIt.data.total);
```
Comment (joefhall) [inline api/routes/projects.js:72]: I think this can be `const` rather than `let` as it isn't reassigned here too?

**Review by joefhall — APPROVED** [review]

**Pull request merged**